File: CS/15_Document/GetViewerPreference.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;
using System.IO;

namespace GetViewerPreference
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Define the input file path for the PDF document
            String input = "..\\..\\..\\..\\..\\..\\Data\\PDFTemplate-Az.pdf";

            // Create a new PdfDocument object to load the PDF file
            PdfDocument doc = new PdfDocument();

            // Read the PDF file from the specified input file path
            doc.LoadFromFile(input);

            // Get the viewer preferences for the loaded PDF document
            PdfViewerPreferences viewer = doc.ViewerPreferences;

            // Create a StringBuilder object to store the details of the viewer preferences
            StringBuilder builder = new StringBuilder();

            // Append a line to the StringBuilder object with the title of the section
            builder.AppendLine("Whether the documents window position is in the center: ");

            // Append a line to the StringBuilder object with the value of the CenterWindow property
            builder.AppendLine("CenterWindow: " + viewer.CenterWindow.ToString());

            // Append a line to the StringBuilder object with the title of the section
            builder.AppendLine("Document displaying mode, i.e. show thumbnails, full-screen, show attachment panel: ");

            // Append a line to the StringBuilder object with the value of the PageMode property
            builder.AppendLine("PageMode: " + viewer.PageMode.ToString());

            // Append a line to the StringBuilder object with the title of the section
            builder.AppendLine("The page layout, i.e. single page, one column: ");

            // Append a line to the StringBuilder object with the value of the PageLayout property
            builder.AppendLine("PageLayout: " + viewer.PageLayout.ToString());

            // Append a line to the StringBuilder object with the title of the section
            builder.AppendLine("Whether window's title bar should display document title: ");

            // Append a line to the StringBuilder object with the value of the DisplayTitle property
            builder.AppendLine("DisplayTitle: " + viewer.DisplayTitle.ToString());

            // Append a line to the StringBuilder object with the title of the section
            builder.AppendLine("Whether to resize the document's window to fit the size of the first displayed page: ");

            // Append a line to the StringBuilder object with the value of the FitWindow property
            builder.AppendLine("FitWindow:" + viewer.FitWindow.ToString());

            // Append a line to the StringBuilder object with the title of the section
            builder.AppendLine("Whether to hide menu bar of the viewer application: ");

            // Append a line to the StringBuilder object with the value of the HideMenubar property
            builder.AppendLine("HideMenubar: " + viewer.HideMenubar.ToString());

            // Append a line to the StringBuilder object with the title of the section
            builder.AppendLine("Whether to hide tool bar of the viewer application: ");

            // Append a line to the StringBuilder object with the value of the HideToolbar property
            builder.AppendLine("HideToolbar: " + viewer.HideToolbar.ToString());

            // Append a line to the StringBuilder object with the title of the section
            builder.AppendLine("Whether to hide UI elements like scroll bars and leave only the page contents displayed: ");

            // Append a line to the StringBuilder object with the value of the HideWindowUI property
            builder.AppendLine("HideWindowUI: " + viewer.HideWindowUI.ToString());

            // Define the output file path for the text file containing the viewer preferences details
            String result = "GetViewerPreference_out.txt";

            // Write the contents of the StringBuilder object to a text file at the specified output file path
            File.WriteAllText(result, builder.ToString());

            //Launch the result file
            DocumentViewer(result);
        }

        private void DocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB/15_Document/GetViewerPreference.vb
Imports System.Text
Imports Spire.Pdf
Imports System.IO

Namespace GetViewerPreference
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Specify the input file path
            Dim input As String = "..\..\..\..\..\..\Data\PDFTemplate-Az.pdf"

            ' Create a new PdfDocument object
            Dim doc As New PdfDocument()

            ' Load the document from the input file
            doc.LoadFromFile(input)

            ' Get the viewer preferences of the document
            Dim viewer As PdfViewerPreferences = doc.ViewerPreferences

            ' Create a StringBuilder object to store the viewer preferences
            Dim builder As New StringBuilder()

            ' Append information about whether the documents window position is in the center
            builder.AppendLine("Whether the document's window position is in the center: ")
            builder.AppendLine("CenterWindow: " & viewer.CenterWindow.ToString())

            ' Append information about the document displaying mode, such as show thumbnails, full-screen, show attachment panel
            builder.AppendLine("Document displaying mode: ")
            builder.AppendLine("PageMode: " & viewer.PageMode.ToString())

            ' Append information about the page layout, such as single page, one column
            builder.AppendLine("The page layout: ")
            builder.AppendLine("PageLayout: " & viewer.PageLayout.ToString())

            ' Append information about whether the window's title bar should display the document title
            builder.AppendLine("Whether the window's title bar should display the document title: ")
            builder.AppendLine("DisplayTitle: " & viewer.DisplayTitle.ToString())

            ' Append information about whether to resize the document's window to fit the size of the first displayed page
            builder.AppendLine("Whether to resize the document's window to fit the size of the first displayed page: ")
            builder.AppendLine("FitWindow: " & viewer.FitWindow.ToString())

            ' Append information about whether to hide the menu bar of the viewer application
            builder.AppendLine("Whether to hide the menu bar of the viewer application: ")
            builder.AppendLine("HideMenubar: " & viewer.HideMenubar.ToString())

            ' Append information about whether to hide the toolbar of the viewer application
            builder.AppendLine("Whether to hide the toolbar of the viewer application: ")
            builder.AppendLine("HideToolbar: " & viewer.HideToolbar.ToString())

            ' Append information about whether to hide UI elements like scroll bars and leave only the page contents displayed
            builder.AppendLine("Whether to hide UI elements like scroll bars and leave only the page contents displayed: ")
            builder.AppendLine("HideWindowUI: " & viewer.HideWindowUI.ToString())

            ' Specify the output file path
            Dim result As String = "GetViewerPreference_out.txt"

            ' Write the contents of the StringBuilder to the output file
            File.WriteAllText(result, builder.ToString())

            ' Close the PDF document
            doc.Close()

            ' Launch the result file
            DocumentViewer(result)
        End Sub

        Private Sub DocumentViewer(ByVal fileName As String)
            Try
                Process.Start(fileName)
            Catch
            End Try
        End Sub
    End Class
End Namespace
